include/aspect/boundary_temperature/ascii_data.h

```diff
@@ -0,0 +1,272 @@
+/*
+  Copyright (C) 2011 - 2014 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#ifndef __aspect__boundary_temperature_ascii_data_h
+#define __aspect__boundary_temperature_ascii_data_h
+
+#include <aspect/boundary_temperature/interface.h>
+#include <aspect/simulator_access.h>
+#include <aspect/utilities.h>
+
+
+namespace aspect
+{
+  namespace BoundaryTemperature
+  {
+    using namespace dealii;
+
+    namespace internal
+    {
+      /**
+       * A base class that implements boundary conditions
+       * determined from a AsciiData input file.
+       */
+      template <int dim>
+      class AsciiDataBoundary : public Utilities::AsciiDataBase<dim>
```

Comment:
I don't care much either way. I commented on this because it was confusing to me that it was in the ```BoundaryTemperature``` namespace. Put it in utilities.h or create a new ascii_data_utilities.h/cc.